src/raftstore/store/fsm/peer.rs

```diff
@@ -341,15 +343,16 @@ impl<'a, T: Transport, C: PdClient> PeerFsmDelegate<'a, T, C> {
         // Commit merge, and then insert a new one to indicate prepare merge
         // is handled.
         let mut meta = self.ctx.store_meta.lock().unwrap();
-        let mark = meta.merge_locks.insert(lock_key, None);
-        if mark.is_some() {
-            mark.unwrap().unwrap().store(true, Ordering::SeqCst);
-            let state = self.fsm.peer.pending_merge_state.as_ref().unwrap();
-            let target_id = state.get_target().get_id();
-            self.ctx
-                .router
-                .force_send(target_id, PeerMsg::Noop(target_id))
-                .unwrap()
+        if let Some(mark) = meta.merge_locks.insert(lock_key, None) {
+            if mark.is_some() {
```

Comment:
When the mark is None?

Reply:
on_ready_prepare_merge() or start() is called twice, seems impossible. And even though it is `None`, we can just ignore it because it doesn't take any effect on `merge_locks`. Or maybe panic here?